### PR TITLE
Lab 2 - Bubble sort

### LINKED ISSUE #166: Lab 2 - Bubble sort
Create an executable go program that returns a sorted copy of int slice using bubble sort:

### PR DESCRIPTION
**Pull request by kasunfdo**
Fixes #166

Review of colleague's PR #155

#### Changes proposed in this PR:
* Implement bubble sort
* Implement insertion sort
* Implement merge sort

### CODE REVIEW

**Review by juliaogris — CHANGES_REQUESTED** [review]
Great work @kasunfdo ! Same [comments](https://github.com/anz-bank/go-course/pull/134#pullrequestreview-243597281) as from Lab4 PR apply.

**Review by juliaogris — CHANGES_REQUESTED** [review]
Nice work!
Please add whose other PR you have reviewed as link to the PR description.
Test some corner cases (empty array, single element array, already sorted array, array in reverse order). You can probably use Table driven tests for this too.

File: 02_bubble/kasunfdo/main.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	out := make([]int, len(s))
```
Comment (jonathanheaden) [inline 02_bubble/kasunfdo/main.go:12]: Why is it needed to create a new slice?

File: 02_bubble/kasunfdo/main.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		if !swapped {
+			break
+		}
+	}
+
+	return out
+}
+
+func insertionSort(input []int) []int {
+	out := make([]int, len(input))
```
Comment (jonathanheaden) [inline 02_bubble/kasunfdo/main.go:33]: Is the new slice needed?

**Review by jonathanheaden — CHANGES_REQUESTED** [review]

File: 02_bubble/kasunfdo/main.go (hunk 1 above)
Comment (jonathanheaden) [reply 02_bubble/kasunfdo/main.go:12]: I see now that it is needed to prevent mutating the original slice

File: 02_bubble/kasunfdo/main.go (hunk 2 above)
Comment (jonathanheaden) [reply 02_bubble/kasunfdo/main.go:33]: I see now that it is needed to prevent mutating the original slice

**Review by jonathanheaden — APPROVED** [review]
looks good

**Review by juliaogris — APPROVED** [review]
LGTM - great work!
